src/RF/e2bp.cpp

```diff
@@ -1,467 +1,466 @@
-                break;
-        }
-    }
-
-    return ret;
-}
-
-// Ask a device to give its status
-DeviceStatus E2bp::pollForStatus() {
-    // Using 0 for 'begin' packet seems to do the trick
-    uint8_t buf[PAYLOAD_LENGTH];
-    reset();
-    setupRFModule();
-    getPayload(buf, PL_STATUS);
-    sendPayload(buf);
-    return firstPayloadStatus;
-}
-
-// Get device mode from previous received data from the device
-DeviceMode E2bp::getDeviceModeFromRecvData() {
-    switch (answerBuf[0]) {
-        case 0:
-            return ON_OFF;
-        case 1:
-            return DIMMER;
-        // YOKIS MVR500ER - 1F - 42 ou 2F - 42
-        case 0x1e:
-        case 0x2f:
-            return SHUTTER;
-        default:
-            return NO_RCPT;
-    }
-}
-
-bool E2bp::press() { return press(false); }
-
-bool E2bp::press(bool dim) {
-    bool ret = true;
-    if (IS_DEBUG_ENABLED) LOG.println("Button pressing");
-    uint8_t buf[PAYLOAD_LENGTH];
-
-    firstPayloadStatus = UNDEFINED;
-    if (dim)
-        getPayload(buf, PL_DIM);
-    else
-        getPayload(buf, PL_BEGIN);
-    ret = sendPayload(buf);
-
-    if (IS_DEBUG_ENABLED) LOG.println("Button pressed");
-    return ret;
-}
-
-bool E2bp::release() {
-    bool ret = true;
-    if (IS_DEBUG_ENABLED) LOG.println("Button releasing");
-    uint8_t buf[PAYLOAD_LENGTH];
-
-    getPayload(buf, PL_END);
-    ret = sendPayload(buf);
-
-    if (IS_DEBUG_ENABLED) LOG.println("Button released");
-    if (ret && device->getMode() == ON_OFF) {
-        device->setStatus(getLastKnownDeviceStatus());
-    } else if (ret && device->getMode() == DIMMER) {
-        device->setStatus(getLastKnownDeviceStatus());
-        device->toggleStatus();
-    }
-    return ret;
-}
-
-// Press and hold a button
-// Send begin payload and dimming payload while
-// release() has not been called
-bool E2bp::pressAndHoldFor(unsigned long duration) {
-    // Only for dimmers and must be > 700ms
-    if (device->getMode() != DIMMER || duration <= 700) return false;
-
-    reset();
-    setupRFModule();
-    press();
-
-    unsigned long timeout = millis() + duration;
-    unsigned long startTime = millis();
-    // Every second until timeout, send a dim command
-    do {
-        // send payload every 1 second
-        if (millis() - startTime >= 1000) {
-            startTime = millis();
-            reset();
-            setupRFModule();
-            press(true);
-        }
-        yield();
-        delayMicroseconds(1000);  // 1 ms delay
-    } while (millis() <= timeout);
-
-    bool ret = release();
-    // no brightness level available but device is ON
-    device->setBrightness(BRIGHTNESS_MAX);
-    return ret;
-}
-
-// Fill a given buffer with the correct payload and return a pointer to it
-uint8_t* E2bp::getPayload(uint8_t* buf, PayloadType type) {
-    buf[0] = 0x00;
-    buf[1] = device->getMode() == SHUTTER ? 0x16 : 0x04;
-    buf[2] = 0x00;
-    buf[3] = 0x20;
-    buf[4] = device->getHardwareAddress()[0];
```

Comment:
That won't work

Reply:
0x16 drives all devices, it's a grouped command.
0x20 doesn't seems to harm.

Reply:
For what I know and reversed, byte number 5 and 6 should contain the hardware address, not 0x00

Reply:
With the scan command it was 0x0000
On: https://github.com/nmaupu/yokis-hack/blob/master/src/RF/e2bp.cpp
You set it and reset it immediately to 0.
That was cosmetics update.

Reply:
buf[4] = device->getHardwareAddress()[0];
    buf[5] = device->getHardwareAddress()[1];
    buf[4] = 0x00;
    buf[5] = 0x00;

Reply:
Ho I see, I completely forgot about that 👍